app/src/ui/dispatcher/dispatcher.ts

```diff
@@ -686,14 +686,24 @@ export class Dispatcher {
 
   /**
    * Delete the branch. This will delete both the local branch and the remote
-   * branch, and then check out the default branch.
+   * branch if includeUpstream is true, and then check out the default branch.
    */
-  public deleteBranch(
+  public deleteLocalBranch(
     repository: Repository,
     branch: Branch,
-    includeRemote: boolean
+    includeUpstream?: boolean
   ): Promise<void> {
-    return this.appStore._deleteBranch(repository, branch, includeRemote)
+    return this.appStore._deleteBranch(repository, branch, includeUpstream)
+  }
+
+  /**
+   * Delete the remote branch.
+   */
+  public deleteRemoteBranch(
+    repository: Repository,
+    branch: Branch
+  ): Promise<void> {
+    return this.appStore._deleteBranch(repository, branch)
```

Comment:
As you know, I'm completely ok with having `deleteLocalBranch` and `deleteRemoteBranch` here (since I'm the one who suggested it in the first place 😂 ), but what I understood from [Markus' last comment](https://github.com/desktop/desktop/pull/11434#discussion_r566194963) is that the dispatcher would also keep its only `deleteBranch` method, and we would only have local/remote versions at the `app/lib/git/branch` level.

I'll leave this up to you 😄

Reply:
I went back and forth on this. I already had it from one of past revisions and now couldn't decide if I want to revert it back. 😄 

We are splitting it out into two concepts already via the `delete-branch-dialog` and `delete-remote-branch-dialog`. 

Because they are so simple, I am inclined to keep the distinction.

Reply:
Sounds good!